uint16 range
uint16 seq
